ReactiveCocoa/Swift/Signal.swift

```diff
@@ -680,11 +680,29 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
+/// errors with `error` on `scheduler`.
+public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
```

Comment:
How about something like:

> If the `interval` is 0, the timeout will be scheduled immediately. The signal must complete synchronously (or on a faster scheduler) to avoid the timeout.